.github/workflows/ci.yml

```diff
@@ -1,116 +1,50 @@
-name: Build/test
+name: CI
+
 on:
   push:
+    branches: [master]
+    paths:
+      - grammar.js
+      - src/**
+      - test/**
+      - bindings/**
+      - binding.gyp
   pull_request:
-    branches:
-      - master
-jobs:
-  changedfiles:
-    runs-on: ubuntu-latest
-    outputs:
-      all: ${{ steps.changes.outputs.all}}
-      c: ${{ steps.changes.outputs.c }}
-      gen: ${{ steps.changes.outputs.gen }}
-    steps:
-      - name: checkout tree-sitter-scala
-        uses: actions/checkout@v4
-        with:
-          fetch-depth: 10
+    paths:
+      - grammar.js
+      - src/**
+      - test/**
+      - bindings/**
+      - binding.gyp
+
+concurrency:
+  group: ${{github.workflow}}-${{github.ref}}
+  cancel-in-progress: true
 
-      - name: Get changed files
-        id: changes
-        run: |
-          echo "all=$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.sha }} | xargs)" >> $GITHUB_OUTPUT
-          echo "c=$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.sha }} | grep '\.\(c\|h\)$' | xargs)" >> $GITHUB_OUTPUT
-          # Generated C code
-          echo "gen=$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.sha }} | grep '\.\(c\|h\)$' | grep -v 'src/scanner.c' | xargs)" >> $GITHUB_OUTPUT
+jobs:
   test:
-    runs-on: ${{ matrix.os }}
-    needs: changedfiles
+    name: Test parser
+    runs-on: ${{matrix.os}}
     strategy:
-      fail-fast: true
+      fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
+        os: [ubuntu-latest, windows-latest, macos-14]
     steps:
-      - name: checkout tree-sitter-scala
+      - name: Checkout repository
         uses: actions/checkout@v4
+      - name: Set up tree-sitter
+        uses: tree-sitter/setup-action/cli@v1
+      - name: Set up examples
+        run: |-
+          git clone https://github.com/scala/scala examples/scala --single-branch --depth=1 --filter=blob:none
+          git clone https://github.com/scala/scala3 examples/scala3 --single-branch --depth=1 --filter=blob:none
```

Comment:
We probably want to pin this to a specific version. Currently we use 2.13.10 and 3.3.0 for example.